heat/core/dndarray.py

```diff
@@ -1471,68 +1471,25 @@ def resplit_(self, axis: int = None):
             self.__lshape_map = None
             return self
 
-        tiles = tiling.SplitTiles(self)
-        new_tile_locs = tiles.set_tile_locations(
-            split=axis, tile_dims=tiles.tile_dimensions, arr=self
+        arr_tiles = tiling.SplitTiles(self)
+        new_tiles = tiling.SplitTiles(self)
+
+        gshape = self.shape
+        new_lshape = list(gshape)
+        new_lshape[axis] = int(arr_tiles.tile_dimensions[axis][self.comm.rank].item())
+
+        recv_buffer = torch.empty(
+            tuple(new_lshape), dtype=self.dtype.torch_type(), device=self.device.torch_device
         )
```

Comment:
Is this still in-place? 😬 Not sure this function ever was in place, really. Expand docs?

What's the  difference in memory usage between `array.resplit_(newaxis)`  and `array = ht.resplit(array, newaxis)`?

Reply:
Well, it is the type of in place resplit as before, which is not really in place for on the ```larray```.  The DNDArray is reused and all the other properties are rewritten, but a new ```larray``` is created. 

We could keep the same larray if the incoming and out coming data has the same length, but it will have to be written in a non-contiguous way, which is not impossible, but not ideal in my opinion.